### ISSUE TITLE
number_separator ignoring double and float numbers

### ISSUE THREAD

**Issue by gianfilippocisternino** [realm/SwiftLint#5043]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I am facing an issue with the `number_separator` rule. SwiftLint is ignoring double and floats numbers and it doesn't trigger any warning nor it autocorrects them.

For example this code:
```
struct Test {
  let anInt = 1000000

  let aDouble: Double = 1000.000000
}
```
Is corrected like this:
```
struct Test {
  let anInt = 1_000_000

  let aDouble: Double = 1000.000000
}
```

### Environment

* SwiftLint version 0.52.2
* Installed using Homebrew

```yml
opt_in_rules:
  - anyobject_protocol
  - array_init
  - attributes
  - closure_spacing
  - comment_spacing
  - conditional_returns_on_newline
  - contains_over_first_not_nil
  - contains_over_range_nil_comparison
  - convenience_type
  - discouraged_none_name
  - empty_count
  - empty_string
  - empty_xctest_method
  - explicit_init
  - fatal_error_message
  - file_header
  - first_where
  - flatmap_over_map_reduce
  - force_unwrapping
  - implicitly_unwrapped_optional
  - joined_default_parameter
  - last_where
  - legacy_multiple
  - let_var_whitespace
  - literal_expression_end_indentation
  - lower_acl_than_parent
  - mark
  - multiline_arguments
  - multiline_arguments_brackets
  - multiline_literal_brackets
  - multiline_parameters
  - multiline_parameters_brackets
  - nimble_operator
  - no_fallthrough_only
  - number_separator
  - object_literal
  - optional_enum_case_matching
  - orphaned_doc_comment
  - overridden_super_call
  - override_in_extension
  - pattern_matching_keywords
  - prefer_self_type_over_type_of_self
  - prefer_zero_over_explicit_init
  - prohibited_super_call
  - quick_discouraged_call
  - quick_discouraged_focused_test
  - quick_discouraged_pending_test
  - reduce_into
  - redundant_nil_coalescing
  - redundant_set_access_control
  - self_in_property_initialization
  - single_test_class
  - sorted_first_last
  - sorted_imports
  - switch_case_on_newline
  - test_case_accessibility
  - unneeded_parentheses_in_closure_argument
  - unowned_variable_capture
  - untyped_error_in_catch
  - unused_closure_parameter
  - vertical_parameter_alignment_on_call
  - vertical_whitespace_between_cases
  - vertical_whitespace_closing_braces
  - yoda_condition

disabled_rules:
  - opening_brace
  - cyclomatic_complexity
  - file_length
  - function_body_length
  - identifier_name
  - nesting
  - type_name
  - trailing_whitespace
  - inclusive_language
 ```

**Comment by SimplyDanny** [comment]
Could you run `echo "let a: Double = 1000.000000" | swiftlint lint --use-stdin --fix` and check if it still doesn't work? For me, the rule triggers and autocorrection works.

**Comment by gianfilippocisternino** [comment]
I did run it and it seems to work. But when I use `swiftlint lint --fix` on a file that contains that code, it says `Done correcting 1 file!` but the file isn't modified.

**Comment by SimplyDanny** [comment]
I really cannot reproduce the issue. Fixes are applied to both literals in your example.

Please make sure you use the latest version of SwiftLint and reduce you configuration. If it works with only the `number_separator` rule enabled, try enabling more rules. Do the same with the number of files that are linted simultaneously. This procedure is not very promising to be honest, but it reduces the noise around the linting process.

**Comment by SimplyDanny** [comment]
Any update here, @gianfilippocisternino?

**Comment by gianfilippocisternino** [comment]
Executing swift `swiftlint lint --fix` on the file specified above now it seems to partially work. The file is fixed from this:
```
struct Test {
  let anInt = 1000000

  let aDouble: Double = 1000.000000
}
```

To this:
```
struct Test {
  let anInt = 1_000_000

  let aDouble: Double = 1_000.000000
}
```

But it's still not correct, since is should fix also the fractional part of the number.

**Comment by SimplyDanny** [comment]
You might need to set the option `minimum_fraction_length` to 0 or anything you like. By default, it is not set so that the rule doesn't trigger on the fractional part of a float.
